src/bun.js/api/BunObject.zig

```diff
@@ -2563,16 +2597,157 @@ pub const Crypto = struct {
 
             const output_digest_slice: []u8 = &output_digest_buf;
 
+            switch (this.*) {
+                .evp => {},
+                .zig => |*inner| {
+                    inner.final(output_digest_slice);
+                    return encoding.encodeWithMaxSize(globalThis, BoringSSL.EVP_MAX_MD_SIZE, output_digest_slice[0..inner.digest_length]);
+                },
+            }
+
             const out = this.evp.final(globalThis.bunVM().rareData().boringEngine(), output_digest_slice);
```

Comment:
can this be moved into the .evp branch?:

Reply:
wanted to keep the diff small for the first iteration to make easier to review and test